lib/endoscope/src/shared.rs

```diff
@@ -0,0 +1,59 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use core::cell::UnsafeCell;
+use core::marker::Sync;
+pub use endoscope_abi::{Shared, State};
+
+#[repr(C)]
+pub struct SharedWrapper {
+    shared: UnsafeCell<Shared>,
+}
+
+unsafe impl Sync for SharedWrapper {}
+
+impl SharedWrapper {
+    pub const fn new() -> Self {
+        SharedWrapper {
+            shared: UnsafeCell::new(Shared {
+                state: State::Preboot as u32,
+                digest: [0xff_u8; 32],
```

Comment:
same as https://github.com/oxidecomputer/hubris/pull/1946/files#r1987774415: get the digest length from the digest type being used

Reply:
done